crates/engine/codegen/src/generation/object/walker.rs

```diff
@@ -272,51 +272,18 @@ impl quote::ToTokens for WalkerFieldMethod<'_> {
                         self.as_ref().#field.iter().copied()
                     }
                 },
-                AccessKind::Ref => quote! {
+                AccessKind::Ref | AccessKind::IdRef => quote! {
                     impl Iter<Item = &'a #ty> + 'a {
                         self.as_ref().#field.iter()
                     }
                 },
-                AccessKind::IdRef => {
-                    if list_as_id_range {
-                        quote! {
-                            impl Iter<Item = &'a #ty> + 'a {
-                                self.#field.walk(self.#ctx)
-                            }
-                        }
-                    } else {
-                        quote! {
-                            impl Iter<Item = &'a #ty> + 'a {
-                                self.as_ref().#field.walk(self.#ctx)
-                            }
-                        }
-                    }
-                }
-                AccessKind::IdWalker => {
-                    if list_as_id_range {
-                        quote! {
-                            impl Iter<Item =  #ty<'a>> + 'a {
-                                self.#field.walk(self.#ctx)
-                            }
-                        }
-                    } else {
-                        quote! {
-                            impl Iter<Item = #ty<'a>> + 'a {
-                                self.as_ref().#field.walk(self.#ctx)
-                            }
+                AccessKind::IdWalker | AccessKind::ItemWalker | AccessKind::RefWalker => {
+                    quote! {
+                        impl Iter<Item =  #ty<'a>> + 'a {
+                            self.as_ref().#field.walk(self.#ctx)
```

Comment:
These changes will result in a few more `as_ref()` calls in the generated code - but I'm not convinced it was worth having all of this code to save a handful of characters.